production.py

```diff
@@ -263,13 +263,18 @@ def do(cls, productions):
             InternalShipment.assign_try(shipments)
 
     def get_cost(self, name):
+        pool = Pool()
+        Uom = pool.get('product.uom')
+
         price = super().get_cost(name)
-        purchase = self.purchase_request and self.purchase_request.purchase
-        if not purchase:
+        line = self.purchase_request and self.purchase_request.purchase_line
+        if not line:
             return price
-
-        return price + purchase.untaxed_amount
-
+        
+        quantity = Uom.compute_qty(
+            self.uom, self.quantity, self.product.default_uom)
+        
+        return price + Decimal(quantity) * line.unit_price
```

Comment:
missing import Decimal()

Reply:
Sorry I forgot to push it, its already fixed.